jablib/src/main/java/org/jabref/logic/citation/SearchCitationsRelationsService.java

```diff
@@ -83,6 +85,21 @@ public List<BibEntry> searchCitations(BibEntry cited) {
         return relationsRepository.readCitations(cited);
     }
 
+    public int getCitationCount(BibEntry citationCounted, Optional<String> actualFieldValue) {
+        Optional<PaperDetails> citationCountResult = null;
+        boolean isFetchingAllowed = relationsRepository.isCitationsUpdatable(citationCounted)
+                || actualFieldValue.isEmpty();
+        if (isFetchingAllowed) {
+            try {
+                citationCountResult = citationFetcher.searchCitationCount(citationCounted);
+            } catch (FetcherException e) {
+                LOGGER.error("Error while fetching citation count for entry", e);
```

Comment:
I think, there also should be a NotificationService.notify (you should have access to dialogService which is a NotificationService)

Reply:
I instead throw the excpetion and handle it in the call back, just let me know if you still want the notify service on the search citation relationService